github_releaser/github_releaser.py

```diff
@@ -225,3 +226,79 @@ def push_files(self, branch_name: str, dest_dir: str, message: str, files: List[
+        """ Downloads specified assets of the release to given output directory
+        Github API: GET /repos/:owner/:repo/releases/tags/:tag
+                    GET /repos/:owner/:repo/releases/assets/:asset_id"""
+
+        with yaspin(text="Getting {} release".format(tag_name)) as spinner:
+
+            url = "{}/repos/{}/{}/releases/tags/{}".format(
+                API_BASEURL, self._account, self._repository, tag_name)
+            response = requests.get(url, auth=self.auth)
+            response_json = response.json()
+
+            if not response.ok:
+                spinner.fail()
+                raise DownloadError(
+                    "Could not find the release {}. {} (Code: {})".format(tag_name, response_json.get("message", None), response.status_code))
+
+            spinner.write("Release is found")
+
+            release_assets = response_json.get("assets")
+            if len(release_assets) == 0:
+                spinner.write("The release doesn't have any assets")
+                spinner.fail()
+                return
+
+            spinner.ok()
+
+        for asset in assets:
+            with yaspin(text="Downloading {}".format(asset)) as spinner:
+                assetId = ""
+                assetName = ""
+                for ra in release_assets:
+                    ra_name = ra.get("name").lower()
+                    matches = 0
+                    if asset == "*" or asset.lower() == ra_name:
+                        matches = 1
+                    elif asset.startswith("*"):
+                        x = asset.replace("*", "")
+                        if ra_name.endswith(x):
+                            matches = 1
+                    elif asset.endswith("*"):
+                        x = asset.replace("*", "")
+                        if ra_name.startswith(x):
+                            matches = 1
+
+                    if matches == 1:
+                        assetId = ra.get("id")
+                        assetName = ra.get("name")
+                        spinner.write(
+                            "Found matching asset with name {}, id: {}. Downloading...".format(assetName, assetId))
+
+                        download_url = "{}/repos/{}/{}/releases/assets/{}".format(
+                            API_BASEURL, self._account, self._repository, assetId)
+                        a_content = requests.get(download_url, auth=self.auth)
+
+                        if not a_content.ok:
+                            spinner.write("Could not download the asset. {} (Code: {})".format(
+                                a_content.get("message", None), a_content.status_code))
+                            spinner.fail()
+                            continue
+
+                        if not path.exists(out_dir):
+                            os.makedirs(out_dir)
+
+                        dest = path.join(out_dir, assetName)
+                        with open(dest, "wb") as f:
```

Comment:
Might need some error handling here, if it can't write the file to disk.

Reply:
Pushed the change, not sure if it's something that you thought of